### PR TITLE
libc/machine/arm: align related implementations of armv7 architecture

### PR DESCRIPTION
**Pull request by anchao**
## Summary

libc/machine/arm: align related implementations of armv7 architecture

1. sync arch elf changes
2. fix cmake compilation break
3. remove the definition of related math files

## Impact

N/A

## Testing

ci-check

### CODE REVIEW

File: libs/libc/machine/arm/armv6-m/arch_elf.c [hunk 1]
```diff
@@ -185,15 +175,29 @@ int up_relocate(const Elf32_Rel *rel, const Elf32_Sym *sym, uintptr_t addr)
     case R_ARM_ABS32:
     case R_ARM_TARGET1:  /* New ABI:  TARGET1 always treated as ABS32 */
       {
-        binfo("Performing ABS32 link at addr=%08lx [%08lx] "
-              "to sym=%p st_value=%08lx\n",
-              (long)addr, (long)(*(uint32_t *)addr), sym,
-              (long)sym->st_value);
+        binfo("Performing ABS32 link "
+              "at addr=%08lx [%08lx] to sym=%p st_value=%08lx\n",
```
Comment (pkarashchenko) [inline libs/libc/machine/arm/armv6-m/arch_elf.c:179]: Why not `"0x08"PRIx32`?

File: libs/libc/machine/arm/armv6-m/arch_elf.c [hunk 2]
```diff
@@ -185,15 +175,29 @@ int up_relocate(const Elf32_Rel *rel, const Elf32_Sym *sym, uintptr_t addr)
...
         *(uint32_t *)addr += sym->st_value;
       }
       break;
 
+#ifdef CONFIG_ARMV7M_TARGET2_PREL
+    case R_ARM_TARGET2:  /* TARGET2 is a platform-specific relocation: gcc-arm-none-eabi
+                          * performs a self relocation */
+      {
+        binfo("Performing TARGET2 link "
+              "at addr=%08lx [%08lx] to sym=%p st_value=%08lx\n",
```
Comment (pkarashchenko) [inline libs/libc/machine/arm/armv6-m/arch_elf.c:192]: same

File: libs/libc/machine/arm/armv6-m/arch_elf.c [hunk 3]
```diff
@@ -216,12 +220,12 @@ int up_relocate(const Elf32_Rel *rel, const Elf32_Sym *sym, uintptr_t addr)
...
          *  1   1   1   1   1   1
-         *  5   4   3   2   1   0   9  8  7  6  5  4  3  2  1  0 Instructions
+         *  5   4   3   2   1   0  9  8  7  6  5  4   3  2  1  0 Instructions
          * +---+------------------------------------------------+
          * |OP |                                                | 32-Bit
          * +---+--+---+---+---+---------------------------------+
-         * |1   1 |J1 | 1 |J2 |              imm11              | BL
-         * +------+---+---+---+---------------------------------+
+         * |1   1 |J1 | 1 |J2 |            imm11                | BL
+         * +------+---+---+---+--------------------------------+
```
Comment (pkarashchenko) [inline libs/libc/machine/arm/armv6-m/arch_elf.c:228]: GitHub preview shows that `-` is needed here. Please make sure that it is well aligned

File: libs/libc/machine/arm/armv6-m/arch_elf.c [hunk 4]
```diff
@@ -351,10 +357,10 @@ int up_relocate(const Elf32_Rel *rel, const Elf32_Sym *sym, uintptr_t addr)
     case R_ARM_MOVW_ABS_NC:
     case R_ARM_MOVT_ABS:
       {
-        binfo("Performing MOVx_ABS [%d] link "
+        binfo("Performing MOVx_ABS [%" PRId32 "] link "
               "at addr=%08lx [%08lx] to sym=%p st_value=%08lx\n",
```
Comment (pkarashchenko) [inline libs/libc/machine/arm/armv6-m/arch_elf.c:361]: same

File: libs/libc/machine/arm/armv8-m/gnu/acle-compat.h [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. The name of the company may not be used to endorse or promote
+ *    products derived from this software without specific prior written
+ *    permission.
```
Comment (hartmannathan) [inline libs/libc/machine/arm/armv8-m/gnu/acle-compat.h:17]: This file's license should be added to LICENSE.

**Review by hartmannathan — CHANGES_REQUESTED** [review]

File: libs/libc/machine/arm/armv8-m/gnu/acle-compat.h (hunk 5 above)
Comment (anchao) [reply libs/libc/machine/arm/armv8-m/gnu/acle-compat.h:17]: Done

File: libs/libc/machine/arm/armv6-m/arch_elf.c (hunk 4 above)
Comment (anchao) [reply libs/libc/machine/arm/armv6-m/arch_elf.c]: Done

File: libs/libc/machine/arm/armv6-m/arch_elf.c (hunk 1 above)
Comment (anchao) [reply libs/libc/machine/arm/armv6-m/arch_elf.c]: Done

File: libs/libc/machine/arm/armv6-m/arch_elf.c (hunk 3 above)
Comment (anchao) [reply libs/libc/machine/arm/armv6-m/arch_elf.c]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
please rebase to the last master which fix the recent ci break

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**